app/controllers/messages_controller.rb

```diff
@@ -107,7 +107,7 @@ def mark
         box_attr = "visible_#{box}"
         user_id_attr = (box == 'inbox') ? 'recipient_id' : 'sender_id'
         Message.where(thread_id: message.thread_id, "#{user_id_attr}": current_user.id).each do |thread_message|
-          thread_message.update(box_attr => false, unread: false)
+          thread_message.update(box_attr => false, unread: false, read_at: Time.zone.now)
```

Comment:
Slightly confused by what's going on with these arguments

Reply:
We had a database column for "what time did the user read this Message" and we weren't actually storing that information when they read it? Possibly because of the threading thing, hmmm. Maybe I need a ||= here.